### DISCUSSION TITLE
Daily account balance in DB

### DISCUSSION THREAD

**Discussion by jadsmrizk** [firefly-iii/firefly-iii#9404] [Firefly III] [answered]
Good day,

I am using firefly iii and have connected the DB to Power BI to generate custom report and dashboards. I could find all needed except the below.

I can't seem to find the balance of accounts on a daily interval for example.

I would like to generate something like the main page of firefly with a bit more granularity, filtering, totals, running average, etc. (Attached below)

![Screenshot_20241028-181829](https://github.com/user-attachments/assets/1e3ccd98-59f3-4912-bf69-205bc93da03a)

Thank you for your suggestions.

**Comment by JC5** [comment]
The charts you're looking for aren't documented yet, but they are available soon! ™️ .

**Reply by jadsmrizk** [reply]
Thank you for the quick reply, I didn't get you well.

I am connecting directly to the postgresql DB and using the data externally. Do you mean there will be a documentation of the Database tables soon?

**Reply by JC5** [reply]
https://docs.firefly-iii.org/explanation/more-information/architecture/

**Reply by jadsmrizk** [reply]
As always, thank you for the responses you are providing.

It seems that the only way to get historized account balance information is in the Transactions table with columns: balance_before and balance_after. For some reason, many lines of these 2 columns are NULL. (I am not sure if I have done something wrong on my side)

I have also seen in Release notes [v6.1.21] the below:
**Add running balance data, see if it can be used in the layout in the future.**

Is this running balance mentionned based on these 2 columns in the Transactions table I have mentionned? or is there a seperate table for that.

**Reply by JC5** [reply, marked as answer]
The running balance columns are fairly new, and are not filled by default. You will have to use the amount column to calculate running balances manually.

You can elect to have Firefly III calculate this for you by setting `USE_RUNNING_BALANCE=true`. Restarting the Firefly III docker container will then populate these columns. This may take a few minutes on the first run.

**Reply by jadsmrizk** [reply]
Very good, thank you
